pkg/api/v1/clients/client_interface.go

```diff
@@ -19,16 +20,46 @@ func DefaultNamespaceIfEmpty(namespace string) string {
 	return namespace
 }
 
+type ResourceWatcher interface {
+	// Deprecated: implemented only by the kubernetes resource client. Will be removed from the interface.
+	Register() error
```

Comment:
I know this is not strictly related to this PR, but could we use the opportunity given by this breaking change to remove the `Register` function from the RC interface? It is a no-op for all RC implementations except for the `kube` one. For the kube client we can move the logic it implements into the constructor.